### ISSUE TITLE
When using `move_to` to go to the same position, a `KeyError` is thrown

### ISSUE THREAD

**Issue by rmellema** [JoelForamitti/agentpy#21]
I was playing around with the package trying to implement a simple sugar scape like environment, and got a confusing `KeyError` when using `move_to` on the `Grid`. Turns out that this happened when an agent tried to `move_to` its current location.

I'm not sure what the expected behaviour should be, but I can see two options that are both better than the current approach:

1. Simply allow it to happen, and have `move_to` do nothing if the position to move to is the same as the current position for the agent. For my current use, this would be fine, but might cause problems for other people
2. Throw a different, but more helpful error. An error like: "Agent {id} is trying to move to its current position" would have saved me some time debugging ;)

I have included a traceback of the error, just in case. But from what I can tell this occurs if the agents try to move to their current position.

```
KeyError                                  Traceback (most recent call last)
<ipython-input-125-dda3ba9185af> in <module>
      1 model = SugarScapeModel({'agents': 20, 'vision': 6, 'max_food': 6, 'spread': 20, 'width': 50, 'height': 50})
----> 2 model.run()

~/Source/SugarScape.py/.env/lib/python3.9/site-packages/agentpy/model.py in run(self, steps, seed, display)
    364         self.sim_setup(steps, seed)
    365         while self.running:
--> 366             self.sim_step()
    367             if display:
    368                 print(f"\rCompleted: {self.t} steps", end='')

~/Source/SugarScape.py/.env/lib/python3.9/site-packages/agentpy/model.py in sim_step(self)
    316         and then calling :func:`Model.step` and :func:`Model.update`."""
    317         self.t += 1
--> 318         self.step()
    319         self.update()
    320         if self.t >= self._steps:

<ipython-input-122-0b943fbda284> in step(self)
     17 
     18     def step(self):
---> 19         self.agents.step()

~/Source/SugarScape.py/.env/lib/python3.9/site-packages/agentpy/sequences.py in __call__(self, *args, **kwargs)
     83 
     84     def __call__(self, *args, **kwargs):
---> 85         return [func_obj(*args, **kwargs) for func_obj in self]
     86 
     87     def __eq__(self, other):

~/Source/SugarScape.py/.env/lib/python3.9/site-packages/agentpy/sequences.py in <listcomp>(.0)
     83 
     84     def __call__(self, *args, **kwargs):
---> 85         return [func_obj(*args, **kwargs) for func_obj in self]
     86 
     87     def __eq__(self, other):

<ipython-input-124-3a612e74c815> in step(self)
     22                 best_pos = (x, y)
     23         print(f"Agent {self.id}, pos {best_pos}")
---> 24         self.scape.move_to(self, best_pos)
     25         self.food = self.scape['food'][best_pos[0]][best_pos[1]]
     26         self.scape['food'][best_pos[0]][best_pos[1]] = 0

~/Source/SugarScape.py/.env/lib/python3.9/site-packages/agentpy/grid.py in move_to(self, agent, pos)
    284             pos = self._border_behavior(pos, self.shape, self._torus)
    285         if self._track_empty:
--> 286             self.empty.replace(pos, pos_old)
    287 
    288         self.grid.agents[pos_old].remove(agent)

~/Source/SugarScape.py/.env/lib/python3.9/site-packages/agentpy/tools.py in replace(self, old_item, new_item)
    134 
    135     def replace(self, old_item, new_item):
--> 136         position = self.item_to_position.pop(old_item)
    137         self.item_to_position[new_item] = position
    138         self.items[position] = new_item

KeyError: (42, 0)
```

**Comment by JoelForamitti** [comment]
Thanks @rmellema! I think it should be allowed, as I can think of many models where that might happen.

**Comment by JoelForamitti** [comment]
Fixed in c0e4c6cf7fdd0db2c8e782b1298b991e01356bda
